### ISSUE TITLE
ASTQyery.cs graph commands detection concerns

### ISSUE THREAD

**Issue by HCRitter** [JustinGrote/MgAqDetect#1]
Hi Justin, I am curious if this method:

return ast.CommandElements[0].Extent.Text.Contains("-mg", OrdinalIgnoreCase);

Is really the best way, I do something similar in my scripts and I try to parse the AST so that I'll get all commands and then check if they are in a graph module on my machine.

I think with your way you would also get some functions which maybe not related to Microsoft.graph if the creator of another function gets creative by choosing the functions name. 

BR
Chris

**Comment by JustinGrote** [comment]
Agreed it might catch another function with an mg prefix, but the alternatives are expensive, such as pre-generating the list of all mg commands and keeping it up to date in the module, or performing a live module lookup for the command and confirm it comes from a Microsoft.Graph module. Neither option is great compared to the low chance of an edge case of simply doing a match on the command.
